### PR TITLE
Decrease swipe sensitivity for hint overlay

### PR DESCRIPTION
**Pull request by prateekbh**
Adds a threshold to swipe before education system is triggered

### CODE REVIEW

File: extensions/amp-story/0.1/amp-story.js [hunk 1]
```diff
@@ -372,8 +372,13 @@ export class AmpStory extends AMP.BaseElement {
     const gestures = Gestures.get(this.element,
         /* shouldNotPreventDefault */ true);
 
-    gestures.onGesture(SwipeXYRecognizer, () => {
-      if (this.bookend_.isActive()) {
+    gestures.onGesture(SwipeXYRecognizer, e => {
+      const xSwipe = Math.abs(e.data.deltaX);
+      const ySwipe = Math.abs(e.data.deltaY);
+      const minSwipeDelta = 50;
```
Comment (alanorozco) [inline extensions/amp-story/0.1/amp-story.js]: Should be a top-level-constant, `MIN_SWIPE_FOR_HINT_OVERLAY_PX  = 50'.

File: extensions/amp-story/0.1/amp-story.js [hunk 2]
```diff
@@ -372,8 +372,13 @@ export class AmpStory extends AMP.BaseElement {
     const gestures = Gestures.get(this.element,
         /* shouldNotPreventDefault */ true);
 
-    gestures.onGesture(SwipeXYRecognizer, () => {
-      if (this.bookend_.isActive()) {
+    gestures.onGesture(SwipeXYRecognizer, e => {
+      const xSwipe = Math.abs(e.data.deltaX);
```
Comment (alanorozco) [inline extensions/amp-story/0.1/amp-story.js]: 1. Refactor into boolean method.
2. Make short-circuit clauses independent.

```javascript
if (this.bookend_.isActive()) {
  return;
}
if (!isSwipeLargeEnoughForHint_(e.data.deltaX, e.data.deltaY) {
  return;
}
```

File: extensions/amp-story/0.1/amp-story.js [hunk 3]
```diff
@@ -372,8 +372,13 @@ export class AmpStory extends AMP.BaseElement {
...
+      const ySwipe = Math.abs(e.data.deltaY);
+      const minSwipeDelta = 50;
+      const canShowEducationOverlay =
```
Comment (alanorozco) [inline extensions/amp-story/0.1/amp-story.js]: Avoid the word `education`  as the current terminology is `hint`.

File: extensions/amp-story/0.1/amp-story.js [hunk 4]
```diff
@@ -372,8 +372,13 @@ export class AmpStory extends AMP.BaseElement {
...
+          xSwipe > minSwipeDelta || ySwipe > minSwipeDelta;
```
Comment (alanorozco) [inline extensions/amp-story/0.1/amp-story.js]: This is inaccurate, you should calculate line distance:
```javascript
Math.sqrt(Math.pow(deltaX, 2) + Math.pow(deltaY, 2)) 
  > MIN_SWIPE_FOR_HINT_OVERLAY_PX``

File: extensions/amp-story/0.1/amp-story.js (hunk 2 above)
Comment (prateekbh) [reply extensions/amp-story/0.1/amp-story.js]: Done

File: extensions/amp-story/0.1/amp-story.js (hunk 1 above)
Comment (prateekbh) [reply extensions/amp-story/0.1/amp-story.js]: Done

File: extensions/amp-story/0.1/amp-story.js (hunk 3 above)
Comment (prateekbh) [reply extensions/amp-story/0.1/amp-story.js]: this is remove

File: extensions/amp-story/0.1/amp-story.js (hunk 4 above)
Comment (prateekbh) [reply extensions/amp-story/0.1/amp-story.js]: oh yeah... makes sense

**Pull request merged**